inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt

```diff
@@ -198,6 +199,21 @@ internal open class InAppMessageBaseView(context: Context, attrs: AttributeSet?)
         }
     }
 
+    // Set close button to black background if the campaign background color is dark.
+    // Brightness is computed based on [Darel Rex Finley's HSP Colour Model](http://alienryderflex.com/hsp.html).
+    // Computed value is from 0 (black) to 255 (white), and is considered dark if less than 130.
+    @SuppressWarnings("MagicNumber")
+    internal fun setCloseButton(button: ImageButton? = null) {
+        val red = Color.red(bgColor)
+        val green = Color.green(bgColor)
+        val blue = Color.blue(bgColor)
+        val brightness = sqrt((red * red * .241) + (green * green * .691) + (blue * blue * .068)).toInt()
+        if (brightness < 130) {
+            (button ?: findViewById(R.id.message_close_button))
+                ?.setImageResource(R.drawable.close_button_white)
+        }
```

Comment:
`else` is not needed because `close_button_black` is the default value?

Reply:
No need since the `close_button_black` is set as default in the layout `xml` files